examples-gcc/peripheral_console_liteos/src/main.c

```diff
@@ -108,34 +76,66 @@ uint32_t query_deep_sleep_allowed(void *dummy, void *user_data)
 {
     (void)(dummy);
     (void)(user_data);
+    return PLATFORM_ALLOW_DEEP_SLEEP;//PLATFORM_ALLOW_DEEP_SLEEP;
     // TODO: return 0 if deep sleep is not allowed now; else deep sleep is allowed
-    return 0;
 }
 
-const char welcome_msg[] = "Built with LiteOS (" HW_LITEOS_KERNEL_VERSION_STRING ")";
+trace_uart_t trace_ctx = {.port = TRACE_PORT};
 
+static const platform_evt_cb_table_t evt_cb_table =
+{
+    .callbacks = {
+        [PLATFORM_CB_EVT_HARD_FAULT] = {
+            .f = (f_platform_evt_cb)cb_hard_fault,
+        },
+        [PLATFORM_CB_EVT_ASSERTION] = {
+            .f = (f_platform_evt_cb)cb_assertion,
+        },
+        [PLATFORM_CB_EVT_HEAP_OOM] = {
+            .f = (f_platform_evt_cb)cb_heap_out_of_mem,
+        },
+        [PLATFORM_CB_EVT_PROFILE_INIT] = {
+            .f = setup_profile,
+        },
+        [PLATFORM_CB_EVT_LLE_INIT] = {
+            .f = on_lle_init,
+        },
+        [PLATFORM_CB_EVT_ON_DEEP_SLEEP_WAKEUP] = {
+            .f = (f_platform_evt_cb)on_deep_sleep_wakeup,
+        },
+        [PLATFORM_CB_EVT_QUERY_DEEP_SLEEP_ALLOWED] = {
+            .f = query_deep_sleep_allowed,
+        },
+        [PLATFORM_CB_EVT_PUTC] = {
+            .f = (f_platform_evt_cb)cb_putc,
+        },
+        [PLATFORM_CB_EVT_TRACE] = {
+            .f = (f_platform_evt_cb)cb_trace_uart,
+            .user_data = &trace_ctx,
+        },
+    }
+};
+
+// TODO: add RTOS source code to the project.
+extern const gen_os_driver_t *os_impl_get_driver(void);
 uintptr_t app_main()
 {
-    platform_set_evt_callback(PLATFORM_CB_EVT_PROFILE_INIT, setup_profile, NULL);
-
-    // setup handlers
-    platform_set_evt_callback(PLATFORM_CB_EVT_HARD_FAULT, (f_platform_evt_cb)cb_hard_fault, NULL);
-    platform_set_evt_callback(PLATFORM_CB_EVT_ASSERTION, (f_platform_evt_cb)cb_assertion, NULL);
-    platform_set_evt_callback(PLATFORM_CB_EVT_HEAP_OOM, (f_platform_evt_cb)cb_heap_out_of_mem, NULL);
-    platform_set_evt_callback(PLATFORM_CB_EVT_ON_DEEP_SLEEP_WAKEUP, on_deep_sleep_wakeup, NULL);
-    platform_set_evt_callback(PLATFORM_CB_EVT_QUERY_DEEP_SLEEP_ALLOWED, query_deep_sleep_allowed, NULL);
-    platform_set_evt_callback(PLATFORM_CB_EVT_PUTC, (f_platform_evt_cb)cb_putc, NULL);
-#ifdef LISTEN_TO_POWER_SAVING
-    platform_set_evt_callback(PLATFORM_CB_EVT_LLE_INIT, on_lle_reset, NULL);
-#endif
+    SYSCTRL_Init();
+    cube_soc_init();
+
+    // setup event handlers
+    platform_set_evt_callback_table(&evt_cb_table);
     setup_peripherals();
+    printf("build@%s\r\n",__TIME__);
+    printf("build@%d\r\n",__LINE__);
 
+    // trace_uart_init(&trace_ctx);
+    // TODO: config trace mask
+    // platform_config(PLATFORM_CFG_TRACE_MASK, 0);
     platform_config(PLATFORM_CFG_POWER_SAVING, PLATFORM_CFG_ENABLE);
+    printf("start up\r\n");
 
     return (uintptr_t)os_impl_get_driver();
 }
 
-void dprintf2(const char *fmt, ...)
-{
-
-}
\ No newline at end of file
+const char welcome_msg[] = "Using Built-in Liteos";
```

Comment:
Please keep the old one: `const char welcome_msg[] = "Built with LiteOS (" HW_LITEOS_KERNEL_VERSION_STRING ")";`